src/arm/arm.c

```diff
@@ -209,58 +209,9 @@ static inline void ARMStep(struct ARMCore* cpu) {
 	LOAD_32(cpu->prefetch[1], cpu->gprs[ARM_PC] & cpu->memory.activeMask, cpu->memory.activeRegion);
 
 	unsigned condition = opcode >> 28;
-	if (condition != 0xE) {
```

Comment:
This will almost certainly cause a performance regression.

Reply:
Hm. I'm not sure I agree. Lemme godbolt it.

Reply:
This pre-condition was added because most ARM opcodes use that, and I actually benchmarked it (several years ago though).

Reply:
Oh, that's what you meant. I thought you meant extracting the switch to a function would. Yeah, I really should have known better, and looking at the godbolt I realize where my mistake was in thinking it was equivalent.

Reply:
Ah, yeah, and combining that with the fact that 0xF needs to be handled specially it probably just makes sense to leave `ARMStep` alone.

Reply:
Rolled back.